### PR TITLE
Fixed HTTP 422 for Github Actions; Improved/added log output

### PR DESCRIPTION
**Pull request by andy-maier**
This PR solved the issue for our repos. It is ready for review and merge. For details, see the commit message.

Note that the description that I added, of how coveralls.io treats the request parameters, is based on experiments. An official confirmation e.g. via discussion of https://github.com/lemurheavy/coveralls-public/issues/1710 is still pending.

Note that this PR does not change the default service name of "github-actions", and I still needed to set the service to "github", either with the `--service=github` option or with the `COVERALLS_SERVICE_NAME: "github"` env var. The question of whether to use service name "github" or "github-actions" is still a bit open I guess. My personal experience is that it always requires "github", but the comment in the code says that "github-actions" was more common.

Note that it is possible for coveralls users to address the issue without this PR, as I described in #252. Nevertheless, I think the issue should be solved in coveralls.

### CODE REVIEW

**Comment by andy-maier** [comment]
I think this PR only fails due to the current failures on Python 3.5 and 3.6 (issue #374).

File: coveralls/api.py [hunk 1]
```diff
@@ -108,15 +108,23 @@ def load_config_from_github(self):
...
         self.config['repo_token'] = os.environ.get('GITHUB_TOKEN')
 
+        # TODO: Clarify default service.
+        # From experiments, it seems that service should always be set to
+        # 'github'. However, that is not yet confirmed - for discussion, see
+        # https://github.com/lemurheavy/coveralls-public/issues/1710.
+        # For compatibility, the service continues to be set to 'github-actions'
+        # for now. Users can specify the service name manually to override
+        # this.
+        service = 'github-actions'
```
Comment (mrexox) [inline coveralls/api.py:118]: Actually service should be `github` to integrate with Github properly.

```suggestion
        service = 'github'
```

File: coveralls/api.py [hunk 2]
```diff
@@ -108,15 +108,23 @@ def load_config_from_github(self):
...
+
+        job = os.environ.get('GITHUB_RUN_ID')
```
Comment (mrexox) [inline coveralls/api.py:120]: Can we change job ID to be `GITHUB_JOB` it should be unique within a workflow, so it's better use it for different jobs.

```suggestion
        job = os.environ.get('GITHUB_JOB')
```

**Review by mrexox — APPROVED** [review]

File: coveralls/api.py (hunk 1 above)
Comment (andy-maier) [reply coveralls/api.py:118]: I agree, I just did not want to change this without being sure. I think that the discussion in https://github.com/lemurheavy/coveralls-public/issues/1710 also argues for using "github".

File: coveralls/api.py (hunk 2 above)
Comment (andy-maier) [reply coveralls/api.py:120]: `GITHUB_JOB` is the **name** of the workflow, not its ID.
 
 `GITHUB_RUN_ID` is an ID that is unique within Actions, but the same for all Actions jobs in the same workflow run. That allows coveralls.io to combine the jobs of a build.

File: coveralls/api.py [hunk 3]
```diff
@@ -276,9 +317,14 @@ def submit_report(self, json_string):
                 new_id = '{}-{}'.format(
                     self.config.get('service_job_id', 42),
                     random.randint(0, sys.maxsize))
-            print('resubmitting with id {}'.format(new_id))
-
             self.config['service_job_id'] = new_id
+            log.warning(
```
Comment (andy-maier) [inline coveralls/api.py:321]: It seems to me that with the fixes in coveralls.io, it is no longer necessary to resubmit, nor would it help to do that with a different service_job_id if the first submission already now sets a service_job_id.

**Comment by TheKevJames** [comment]
@andy-maier thank you for going down the GHA rabbithole and figuring this all out. I've just read through this PR, the linked coveralls-public details, etc and done some testing. I think this is the right track (setting both values the same), as is using `github` instead of `github-actions` by default and avoiding the retry-on-422. I've created #427 with those changes as well as yours here merged together, and plan to merge and deploy it shortly as a new major version.

I did notice that despite the `job_id == number` you have in this PR, in [the successful config you've got referenced by the coveralls-public issue](https://github.com/zhmcclient/python-zhmcclient/commit/38377ce56ae16e2c1157efd38e4e43ac411116c3), you're doing something different. Could you comment on that? I'm wondering if that's a default we should adopt?

Also curious on your change to add the `omit` value, what is that for?

**Pull request closed without merging**